src/cider/exec/nextgen/context/RuntimeContext.h

```diff
@@ -29,12 +29,16 @@
 namespace cider::exec::nextgen::context {
 class RuntimeContext {
  public:
-  RuntimeContext(int64_t ctx_num) : runtime_ctx_pointers_(ctx_num, nullptr) {}
+  RuntimeContext(int64_t ctx_num) : runtime_ctx_pointers_(ctx_num, nullptr) {
+    string_heap_ptr_ = std::make_shared<StringHeap>();
```

Comment:
Maybe the string heap should initialize during codegen if need? like call `runtime_ctx.setStringHeap()`.

Reply:
make sense, updated. Please take a look again.